docs/network-layout-spec.md

```diff
@@ -0,0 +1,172 @@
+id: testnet_abcd
+
+# Configuration relating to monitoring of the Tendermint network nodes. Right
+# now, the idea is to support SignalFX and/or InfluxDB. At least one of the two
+# must be configured.
+monitoring:
+  # Any SignalFX-related configuration parameters will go here.
+  signalfx:
+    api_token: $SIGNALFX_API_TOKEN
+    realm: $SIGNALFX_REALM
+
+  influxdb:
+    # Set to `yes` to deploy the Grafana/InfluxDB monitoring server. If set to
+    # `no`, you will need to supply InfluxDB details to which to send the
+    # metrics.
+    deploy: yes
+
+    # Where do you want to deploy the monitoring server? (Can also use
+    # notation: `us_east_1`)
+    region: us-east-1
+
+    # If deploy is set to `no`, the InfluxDB URL is required.
+    #url: https://your-own-influxdb-host:8086
+    #database: tendermint
+    #username: tendermint
+
+    # What password should we use for InfluxDB? If not specified, a strong
+    # password will automatically be generated.
+    password: $INFLUXDB_PASSWORD
+
+# A mapping of named sub-groups of nodes within the desired test network
+# resource group.
+tendermint_network:
+  # `my_validators` could be any kind of unique identifier for this group
+  - my_validators:
+      # If you want to deploy an official release of Tendermint, just specify
+      # a version number here and its binary will be deployed from GitHub.
+      tendermint: v0.31.7
+
+      # Are these nodes' details to be included in the `genesis.json` file?
+      # (Default: yes)
+      in_genesis: yes
+
+      # Should these nodes' Tendermint services be started? (Default: yes)
+      #start: yes
+
+      # Where to find the configuration file to use as a template for
+      # generating configuration for all of the nodes in this sub-network.
+      config_template: ./my-validators-config.toml
+
+      # The group name(s) for nodes to consider as seeds
+      use_seeds:
+        - my_seeds
+
+      # To prepopulate persistent peers' addresses, specify a list of group
+      # names here. Can specify this group's own name here.
+      #persistent_peers:
+      #  - my_validators
+
+      # Where to deploy nodes for this group. Note that nodes are numbered
+      # according to the order in which the regions appear here.
+      regions:
+        - us_east_1: 2    # my_validators[0], my_validators[1]
+        - us_east_2: 1    # my_validators[2]
+        - us_west_1: 3    # my_validators[3], my_validators[4], my_validators[5]
+
+  - my_seeds:
+      tendermint: v0.31.7
+      config_template: ./seednode-config.toml
+      # We want these nodes to be seeds
+      seeds: yes
+      # Where to deploy seed nodes for this group
+      regions:
+        - us_east_1: 1    # my_seeds[0]
+        - us_west_1: 1    # my_seeds[1]
+
+  - late_joiners:
+      # If you want to deploy a custom Tendermint binary, specify its path
+      # instead of an official release version.
+      tendermint: /path/to/local/tendermint
```

Comment:
In general for the 'tendermint' entries I assume we can point to a version or local image, right?

Reply:
Yes, my assumption here is that if the string begins with `v`, it's a specific version we want to deploy (in which case it'll go fetch the binary from GitHub), and otherwise it will assume that it's a filesystem path and will push that binary from the local filesystem to the remote nodes.